test/e2e/es/license_test.go

```diff
@@ -89,21 +90,17 @@ func TestEnterpriseTrialLicense(t *testing.T) {
 	licenseBytes, err := ioutil.ReadFile(test.Ctx().TestLicense)
 	require.NoError(t, err)
 
+	k := test.NewK8sClientOrFatal()
+
 	esBuilder := elasticsearch.NewBuilder("test-es-trial-license").
 		WithESMasterDataNodes(1, elasticsearch.DefaultResources)
 
-	var licenseTestContext elasticsearch.LicenseTestContext
+	licenseTestContext := elasticsearch.NewLicenseTestContext(k, esBuilder.Elasticsearch)
```

Comment:
nit:  I think we can avoid `k` as it is only used in this statement.
```suggestion
		licenseTestContext := elasticsearch.NewLicenseTestContext(test.NewK8sClientOrFatal(), esBuilder.Elasticsearch)
```